src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -349,12 +355,72 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
         trigger=TriggerUpdateParameters(
             source_triggers=source_trigger_update_params,
             base_image_trigger=base_image_trigger_update_params
+        ),
+        credentials=get_custom_registry_credentials(
+            cmd,
+            auth_mode=auth_mode
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_credential_add(cmd,
+                            client,
+                            task_name,
+                            registry_name,
+                            login_server,
+                            username,
+                            password,
+                            resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters = cmd.get_models('TaskUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        credentials=get_custom_registry_credentials(
+            cmd,
+            login_server=login_server,
+            username=username,
+            password=password
         )
     )
 
     return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
 
 
+def acr_task_credential_delete(cmd,
+                               client,
+                               task_name,
+                               registry_name,
+                               login_server,
+                               resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters = cmd.get_models('TaskUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        credentials=get_custom_registry_credentials(
+            cmd,
+            login_server=login_server
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_credential_list(cmd,
+                             client,
+                             task_name,
+                             registry_name,
+                             resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    resp = client.get_details(resource_group_name, registry_name, task_name).credentials
+    return None if resp is None else resp.custom_registries
```

Comment:
Is `custom_registries` a dict? If so, `return {}`, also why is the server returning `null` instead of empty dict? /cc @mnltejaswini

Reply:
The server returns `null` if there are no credentials. I think in general in az commands if you list anything and it does not have anything, you don't get any response (`None`).

Reply:
cc @mnltejaswini @northtyphoon